swarm/api/storage.go

```diff
@@ -42,10 +42,11 @@ func NewStorage(api *Api) *Storage {
 //
 // DEPRECATED: Use the HTTP API instead
 func (self *Storage) Put(content, contentType string) (string, error) {
-	key, _, err := self.api.Put(content, contentType)
+	key, wait, err := self.api.Put(content, contentType)
```

Comment:
if we don't want to make `Storage.Put` synchronous, we can hook to the wait function with a global test var (as @janos suggested). However I am not sure why we want to make `Storage.Put` asynchronous in general (with respect to local memory, and disk commit), when a user could just do `go api.Put(...)` if they want to have it asynchronous.

Reply:
OK.

In this case the API of deprecated Storage.Put is not useful, as we lose the `wait` function and have to guess when the commit to memory/disk happened. Since this method is barely used and also deprecated, I just wrapped the `self.api.Put` with it, and returned the wait function further up the stack.